lib/routes/home.dart

```diff
@@ -0,0 +1,40 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'route_names.dart';
+import '../utils/favorite_locations.dart';
+import '../views/location_card.dart';
+
+class Home extends StatelessWidget {
+  const Home({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('World Time'),
+        actions: [
+          IconButton(
+            icon: const Icon(Icons.list),
+            onPressed: () {
+              Navigator.pushNamed(context, setFavoriteLocations);
+            },
+            tooltip: 'Saved Suggestions',
+          ),
+        ],
+      ),
+      body: Consumer<FavoriteLocations>(
+        builder: (context, favoriteLocations, child) {
+          return favoriteLocations.currentLocations.isNotEmpty
+              ? PageView(
```

Comment:
Consider using the `PageView.builder` constructor here: https://master-api.flutter.dev/flutter/widgets/PageView/PageView.builder.html

The builder constructor lets you instantiate the `LocationCard`s lazily when they are needed instead of instantiating them all upfront even if the user never swipes to them to show them.